ecoli/processes/metabolism_redux.py

```diff
@@ -307,37 +288,32 @@ def next_update(self, timestep, states):
         total_maintenance = flux_gam + flux_ngam + flux_gtp
         maintenance_target = total_maintenance.asNumber()
 
-        # binary kinetic targets
-        reaction_catalyst_counts = np.array([sum([current_catalyst_counts[
-            enzyme_idx] for enzyme_idx in enzymes_idx]) for enzymes_idx
-            in self.catalyzed_rxn_enzymes_idx])
+        # binary kinetic targets - sum up enzyme counts for each reaction
+        reaction_catalyst_counts = np.array([sum([current_catalyst_counts[enzyme_idx]
+                                                  for enzyme_idx in enzymes_idx])
+                                             if len(enzymes_idx) > 0 else -1
```

Comment:
Doesn't this do the same thing as before? Instead of filtering for reactions that are enzyme-catalyzed (`len(enzymes_idx) > 0`) in the `__init__`, you just do it here at every `next_update`. Before, both `self.catalyzed_rxn_enzymes_idx` and `self.enzyme_catalyzed_rxn_idx` contained only enzyme and reaction indices where `len(enzymes_idx) > 0`. In that case, I think the same indices should be retrieved for `binary_kinetic_idx`.